### PR TITLE
Android: Handle URL links

### PR DESCRIPTION
**Pull request by fstanis**
Relevant proposal: #5168

This reuses the parsing logic from #5302 and adds support for Android. The easiest way to test is via adb:

```
$ adb shell am start -a android.intent.action.VIEW -d 'joplin://x-callback-url/openNote?id=<note id>'
```

### CODE REVIEW

File: packages/app-mobile/root.tsx [hunk 1]
```diff
@@ -822,6 +826,60 @@ class AppComponent extends React.Component {
 		}
 	}
 
+	private async openCallbackUrl(url: string) {
```
Comment (laurent22) [inline packages/app-mobile/root.tsx:829]: Could you move this function to a separate file please?

File: packages/app-mobile/root.tsx [hunk 2]
```diff
@@ -822,6 +826,60 @@ class AppComponent extends React.Component {
...
+			if (!await Note.load(params.id)) {
+				reg.logger().error(`note ${params.id} doesn't exist`);
+				return;
+			}
+			await this.prepareToNavigate();
+			await this.props.dispatch({
+				type: 'NAV_GO',
+				noteId: params.id,
+				routeName: 'Note',
+			});
```
Comment (laurent22) [inline packages/app-mobile/root.tsx:847]: Could you refactor to avoid the prepareToNavigate/dispatch repetition?

Comment (fstanis) [reply packages/app-mobile/root.tsx]: Done

File: packages/app-mobile/root.tsx (hunk 1 above)
Comment (fstanis) [reply packages/app-mobile/root.tsx]: Done

**Comment by kominoshja** [comment]
#5571 seems to be stuck on this, yet this seems to pass all checks, friendly nudge @laurent22 @fstanis

**Comment by laurent22** [comment]
Closed due to https://github.com/laurent22/joplin/pull/5571#issuecomment-1361108479

**Pull request closed without merging**